src/testdrive/src/action/shell.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use async_trait::async_trait;
+
+use crate::action::{Action, State};
+use crate::parser::BuiltinCommand;
+
+use tokio::process::Command;
+
+pub struct ExecuteAction {
+    command: String,
+}
+
+pub fn build_execute(cmd: BuiltinCommand) -> Result<ExecuteAction, String> {
+    Ok(ExecuteAction {
+        command: cmd.input.join("\n"),
+    })
+}
+
+#[async_trait]
+impl Action for ExecuteAction {
+    async fn undo(&self, _: &mut State) -> Result<(), String> {
+        Ok(())
+    }
+
+    async fn redo(&self, _: &mut State) -> Result<(), String> {
+        println!("$ shell-execute\n{}", self.command);
+
+        let status = Command::new("bash")
+            .arg("-c")
+            .arg(&self.command)
+            .spawn()
+            .expect("failed to spawn")
```

Comment:
This will crash the process, is this what you meant? If not, `map_err(|e| e.to_string())?` will bubble it up

Reply:
Thanks.